qa/src/test/java/io/camunda/zeebe/process/test/qa/ExtendAbstractsTest.java

```diff
@@ -15,80 +15,76 @@
  */
 package io.camunda.zeebe.process.test.qa;
 
-import com.google.common.reflect.ClassPath;
 import io.camunda.zeebe.process.test.extension.ZeebeProcessTest;
-import java.io.IOException;
-import java.lang.annotation.Annotation;
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-import java.util.Optional;
+import io.github.classgraph.ClassGraph;
+import io.github.classgraph.ClassInfo;
+import io.github.classgraph.ClassInfoList;
+import java.util.stream.Stream;
 import org.assertj.core.api.SoftAssertions;
-import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
 
 class ExtendAbstractsTest {
 
-  public static final String EMBEDDED = "embedded";
-  public static final String TESTCONTAINER = "testcontainer";
   public static final Class<ZeebeProcessTest> EMBEDDED_ANNOTATION = ZeebeProcessTest.class;
   public static final Class<io.camunda.zeebe.process.test.extension.testcontainer.ZeebeProcessTest>
       TESTCONTAINER_ANNOTATION =
           io.camunda.zeebe.process.test.extension.testcontainer.ZeebeProcessTest.class;
-  private static final String BASE_PACKAGE = "io.camunda.zeebe.process.test.qa";
-  private static final String ABSTRACTS = "abstracts";
+  private static final String ABSTRACT_PACKAGE = "io.camunda.zeebe.process.test.qa.abstracts";
+  private static final String EMBEDDED_PACKAGE = "io.camunda.zeebe.process.test.qa.embedded";
+  private static final String TESTCONTAINER_PACKAGE =
+      "io.camunda.zeebe.process.test.qa.testcontainer";
 
-  @Test
-  void testAbstractClassesAreExtendedWithBothExtensions() throws IOException {
-    final Map<String, List<Class<?>>> classes = new HashMap<>();
-    classes.put(ABSTRACTS, new ArrayList<>());
-    classes.put(EMBEDDED, new ArrayList<>());
-    classes.put(TESTCONTAINER, new ArrayList<>());
+  private static final ClassInfoList EXTENDING_CLASSES =
```

Comment:
:wrench: Not really needed. You put two packages into one bucket and then later filter out by package. You could as well prepare two buckets.